synapse/rest/media/v1/preview_html.py

```diff
@@ -323,10 +323,13 @@ def _iterate_over_text(
 def rebase_url(url: str, base: str) -> str:
```

Comment:
I think this function effectively:
- returns `url` unmodified if it is a fully specified, absolute URL
- otherwise resolves `url` as a relative URL against `base` and returns the resolved result

If so, a suggested docstring:

```python
"""
Resolves a potentially relative `URL` against an absolute `base` URL.

For example:
    >>> rebase_url("subpage", "https://example.com/foo/")
    'https://example.com/hello/subpage'
    >>> rebase_url("sibling", "https://example.com/foo")
    'https://example.com/sibling'
    >>> rebase_url("/hello", "https://example.com/foo/")
    'https://example.com/hello'
    >>> rebase_url("https://alice.com/a/", "https://bob.com/b/")
    'https://alice.com/a'
"""
```

Reply:
Looks like we don't have tests for this function either, I'll add some.